### PR TITLE
Fix more Test classes to work with segrep

### PR DESCRIPTION
**Pull request by Rishikesh1159**
### Description
[Describe what this change achieves]

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/src/internalClusterTest/java/org/opensearch/action/bulk/BulkWithUpdatesIT.java [hunk 1]
```diff
@@ -79,6 +80,7 @@
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.nullValue;
 
+@LuceneTestCase.AwaitsFix(bugUrl = "Needs more deep dive")
```
Comment (dreamer-89) [inline server/src/internalClusterTest/java/org/opensearch/action/bulk/BulkWithUpdatesIT.java:83]: May be let these tests run ? We will get more data around failure

File: server/src/test/java/org/opensearch/search/aggregations/bucket/ShardSizeTestCase.java [hunk 2]
```diff
@@ -46,6 +46,7 @@
 import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertSearchResponse;
 import static org.hamcrest.Matchers.is;
 
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST)
```
Comment (dreamer-89) [inline server/src/test/java/org/opensearch/search/aggregations/bucket/ShardSizeTestCase.java:49]: This is not needed ? May be we can add in child classes if possible

File: test/framework/src/main/java/org/opensearch/snapshots/AbstractSnapshotIntegTestCase.java [hunk 3]
```diff
@@ -473,7 +473,11 @@ protected void indexRandomDocs(String index, int numdocs) throws InterruptedExce
         }
         indexRandom(true, builders);
         flushAndRefresh(index);
-        assertDocCount(index, numdocs);
+        try {
+            assertBusy(() -> { assertDocCount(index, numdocs); });
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
```
Comment (dreamer-89) [inline test/framework/src/main/java/org/opensearch/snapshots/AbstractSnapshotIntegTestCase.java:480]: 👍

**Review by dreamer-89 — CHANGES_REQUESTED** [review]

File: server/src/internalClusterTest/java/org/opensearch/action/bulk/BulkWithUpdatesIT.java (hunk 1 above)
Comment (Rishikesh1159) [reply server/src/internalClusterTest/java/org/opensearch/action/bulk/BulkWithUpdatesIT.java:83]: Sure, will remove this

File: server/src/test/java/org/opensearch/search/aggregations/bucket/ShardSizeTestCase.java (hunk 2 above)
Comment (Rishikesh1159) [reply server/src/test/java/org/opensearch/search/aggregations/bucket/ShardSizeTestCase.java:49]: Sure, will update this

**Review by dreamer-89 — APPROVED** [review]

**Pull request merged**